packages/application-generic/src/dtos/process-subscriber-job.dto.ts

```diff
@@ -0,0 +1,29 @@
+import { ISubscribersDefine, ITenantDefine } from '@novu/shared';
+import { SubscriberEntity } from '@novu/dal';
+
+import {
+  IBulkJobParams,
+  IJobParams,
+} from '../services/queues/queue-base.service';
+
+export interface IProcessSubscriberDataDto {
+  environmentId: string;
+  organizationId: string;
+  userId: string;
+  transactionId: string;
+  identifier: string;
+  payload: any;
+  overrides: Record<string, Record<string, unknown>>;
+  tenant?: ITenantDefine | null;
+  actor?: SubscriberEntity;
+  subscriber: ISubscribersDefine;
+  templateId: string;
+}
+
+export interface IProcessSubscriberJobDto extends IJobParams {
+  data?: IProcessSubscriberDataDto;
+}
+
+export interface IProcessSubscriberBulkJobDto extends IBulkJobParams {
+  data: IProcessSubscriberDataDto;
```

Comment:
This is part of the complexity i was talking about in the issue description, due to the metrics add execution allowing undefined data objects we need to create two separate interfaces because we do not want to allow undefined in the data as much as possible e.x the bulk DTO.